awswrangler/redshift.py

```diff
@@ -238,7 +262,23 @@ def _create_table(
     s3_additional_kwargs: Optional[Dict[str, str]] = None,
 ) -> Tuple[str, Optional[str]]:
     if mode == "overwrite":
-        _drop_table(cursor=cursor, schema=schema, table=table)
+        if overwrite_method == "truncate":
+            try:
+                # Truncate commits current transaction, if successful.
+                # Fast, but not atomic.
+                _truncate_table(cursor=cursor, schema=schema, table=table)
+            except redshift_connector.error.ProgrammingError as e:
+                # Caught "relation does not exist".
+                _logger.debug(str(e))
+                con.rollback()
+            _begin_transaction(cursor=cursor)
```

Comment:
Can you help me understand this part? If the truncate fails because of the table/view does not exist, we roll back and then we begin the transaction regardless? What does begin transaction achieves?

Reply:
Exactly. `TRUNC` fails if there is no table so except catches that, and rolls back the transaction. If `TRUNC` succeeds, it [commits the current transaction](https://docs.aws.amazon.com/redshift/latest/dg/r_TRUNCATE.html) so ether way we have to begin a new one.

Reply:
Now that I think about it, it is probably worth to:
1) Check the error message for the exact code of `relation ... does not exist` error.
2) Put `_begin_transaction` into the `finally` block just in case something in `except` goes south.
Would you agree?

Reply:
Understood, thanks. I agree with catching the exact message and putting the begin transaction in a finally block

Reply:
FYI I changed my mind on the latter - actually, we *don't* want to begin transaction if something in except block fails - that would just be a waste of transaction. Late night replies 😅